### ISSUE TITLE
Visual composer compatibility

### ISSUE THREAD

**Issue by Dahnark** [restrictcontentpro/restrict-content-pro#1481]
At this moment I know that your plugin is compatible with Visual Composer, but a lot of times Visual Composer add multiple shortcodes around your shortcodes and it is tedious to have to delete a lot of times these useless shortcodes to get the page working without front end shortcodes.

Can you add row/column/element compatibility with Visual Composer?

**Comment by mindctrl** [comment]
@Dahnark we've reached out to the VC folks a few times in the past, but we never hear back. Can you show me an example of what you mean by it wrapping multiple shortcodes around our shortcode?

**Comment by Dahnark** [comment]
@mindctrl if you have this in classic view and you publish it, the web is going to be fine:

```
shortcode (hide to logged users)
       shortcode row
       shortcode end row
shortcode end (hide to logged users)

```
But now you suppose I want to edit my page with visual composer, after edit for example the first shortcode row, visual composer add custom code around your shortcodes:

```
shortcode row (generated automatically by vc)
       shortcode (hide to logged users)
shortcode row end (generated automatically by vc)
             shortcode row
             shortcode end row
shortcode row (generated automatically by vc)
       shortcode end (hide to logged users)
shortcode row end (generated automatically by vc)

```
Then now with this automatic shortcodes inserted, the hide to logged users shortcode starts to hide from here `shortcode row end (generated automatically by vc)` to `shortcode row (generated automatically by vc)`. Then the first and last shortcodes are out of the hide to logged users and appears in web like plain text.

This problem is more tedious than above if you want to show a row to logged users and hide this row to not logged users, and another row hide to logged users and show this row to not logged users.

Sorry if my explanation is not very good but it is difficult to explain.

**Comment by mindctrl** [comment]
This looks more like a VC bug than anything related to Restrict Content Pro. We've talked about an integration with their proprietary system, but haven't decided to do it or allotted any time to it. The last time I looked, there was no good developer documentation that I could find, so it would involve digging through their code to see if/where we can tie in. We'll open a new issue if we decide to add an official integration.
